pkg/executables/cmk.go

```diff
@@ -120,6 +120,38 @@ func (c *Cmk) ValidateServiceOfferingPresent(ctx context.Context, zoneId string,
 	return nil
 }
 
+func (c *Cmk) ValidateDiskOfferingPresent(ctx context.Context, zoneId string, diskOffering v1alpha1.CloudStackResourceDiskOffering) error {
+	command := newCmkCommand("list diskofferings")
+	if len(diskOffering.Id) > 0 {
+		applyCmkArgs(&command, withCloudStackId(diskOffering.Id))
+	} else {
+		applyCmkArgs(&command, withCloudStackName(diskOffering.Name))
+	}
+	applyCmkArgs(&command, withCloudStackZoneId(zoneId))
```

Comment:
do we need to take "account" or "domain" in this command?

Reply:
Good question. 
I used our cmk tool to test the list diskofferings and found account was not an option. And if I applied domainId filter, nothing returned, even we do have valid disk offerings under specified domain. So I believe cloudstack's API list diskoffering implementation for domainId/Account is not working.

https://cloudstack.apache.org/api/apidocs-4.14/apis/listDiskOfferings.html

```(localcloud) 🐱 > list diskofferings domainid=5300cdac-74d5-11ec-8696-c81f66d3e965 
(localcloud) 🐱 > list diskofferings 
{
  "count": 4,
  "diskoffering": [
    {
      "created": "2022-01-13T20:01:51-0500",
      "disksize": 5,
      "displayoffering": true,
      "displaytext": "Small Disk, 5 GB",
      "id": "399f1b0d-742f-4474-a138-7932b9a54fed",
      "iscustomized": false,
      "name": "Small",
      "provisioningtype": "thin",
      "storagetype": "shared"
    },
    {
      "created": "2022-01-13T20:01:51-0500",
      "disksize": 20,
      "displayoffering": true,
      "displaytext": "Medium Disk, 20 GB",
      "id": "9c4a12f6-4cc5-470a-98b6-cf40306f0e51",
      "iscustomized": false,
      "name": "Medium",
      "provisioningtype": "thin",
      "storagetype": "shared"
    },
    {
      "created": "2022-01-13T20:01:51-0500",
      "disksize": 100,
      "displayoffering": true,
      "displaytext": "Large Disk, 100 GB",
      "id": "3a9f4649-67e3-4af8-8a9c-37a582383217",
      "iscustomized": false,
      "name": "Large",
      "provisioningtype": "thin",
      "storagetype": "shared"
    },
    {
      "created": "2022-01-13T20:01:51-0500",
      "disksize": 0,
      "displayoffering": true,
      "displaytext": "Custom Disk",
      "id": "df36c956-9f4a-4cc3-84c9-239e0914d0f8",
      "iscustomized": true,
      "name": "Custom",
      "provisioningtype": "thin",
      "storagetype": "shared"
    }
  ]
}
(localcloud) 🐱 > list diskofferings 
domainid=    filter=      id=          isrecursive= keyword=     listall=     name=        page=        pagesize=    zoneid=  
```